PR - Add project feature

Tom-Sarfo opened the pull request:
- The Add Project feature allows users to create a project by adding team members and providing a name for the project
- This will allow users to keep track of tasks for a unique project.

- Made use of React router dom for page navigation
- Made use of  React Context API and the Reducer hook for state management

Tom-Sarfo committed:
initial commit

Tom-Sarfo committed:
Incluede add project Layout files

Tom-Sarfo committed:
Included add project layout files

Tom-Sarfo committed:
Included routing to project layout files

Tom-Sarfo committed:
Included a form to create Project
- added a context for state stateManagement
- included a reducer function to dispatch project acitons
- added a nested form for adding member to project

Tom-Sarfo committed:
Hooked up Context, reducer function in app entry
- added ProjectBoard to display CreateProjectForm
- modified app.js to include routerProvider

Tom-Sarfo committed:
Added a way to list project after form submit
Added a way to list project after form submit

Tom-Sarfo committed:
Added route parameter to link project to tasks
Added routeError page for nice error UX

Tom-Sarfo committed:
Added route parameter to link project to tasks
Added routeError page for nice error UX

noramens commented on src/ProjectLayout/CreateProjectForm.js:
```diff
@@ -0,0 +1,51 @@
+  const dispatch = useContext(ProjectDispatch)
+  const [projectName, setProjectName] = useState('');
+  const [ members, setMembers ] = useState([])
+  const navigate = useNavigate()
```
@Tom-Sarfo its more elegant to separate your other hooks from your states

Tom-Sarfo replied:
Sure

noramens commented on src/ProjectLayout/MemberForm.js:
```diff
@@ -0,0 +1,42 @@
+    
+  return (
+    <div>
+      <form onSubmit={e => handleSubmit(e)}>
```
@Tom-Sarfo writing your code this way would have still sufficed:

`<form onSubmit={handleSubmit}>`

noramens commented on src/ProjectLayout/MemberForm.js:
```diff
@@ -0,0 +1,42 @@
+        <button type='submit' id='add-member-btn'>Add</button>
+      </form>
+      <div className='chips-container'>
+        { members.map(myMember => (
```
@Tom-Sarfo you should check for the availability of a variable before mapping on it. You can use optional chaining in this scenario.

Tom-Sarfo replied:
Okay sure.

noramens commented on src/ProjectLayout/ProjectBoard.js:
```diff
@@ -0,0 +1,24 @@
+        </Link>
+    </main>
+  )
+}
```
@Tom-Sarfo consider using `Prettier` extension on vscode to help automatically format your code when you save it

Tom-Sarfo replied:
Okay
I have it installed though. I will configure it to format onsave

Tom-Sarfo committed:
Reviewed changes made
- congigured prettier onsave
- formated all files with prettier
- added optional chaining operator on all mapped variables
- separated state hooks from other hooks
